tests/gordo_components/server/test_anomaly_view.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+import numpy as np
+
+import tests.utils as tu
+
+
+@pytest.mark.parametrize(
+    "data_to_post",
+    [
+        {
+            "X": np.random.random(size=(10, len(tu.SENSORS_STR_LIST))).tolist()
+        },  # Nested records
+        {
+            "X": np.random.random(size=len(tu.SENSORS_STR_LIST)).tolist()
+        },  # Single records 1d
+        None,  # No data, use GET
+    ],
+)
+def test_anomaly_prediction_endpoint(
```

Comment:
Generally, I feel this:
* We no longer support "non-pipeline" models. We should explicitly show this in tests? With something failing?
* Maybe we should have a pipeline with only one step? I think it is the `trained_model_directory` in conftest which is the source for many of the "end-to-end" (TM) tests.

Could we do some parametrization or maybe just explicitly have a couple of different `trained_model_directory` versions (no pipeline, will fail in the server region, due to .steps and stuff, and single step, and more steps (like today))?

Reply:
Check out the latest tests, and see if that doesn't suit your fancy. Thought about parametrizing the tests but I think this route it's quicker (both test timing and less complex tests) and is what the server  uses anyway to interact with the model `ModelMixin`